core/src/com/unciv/logic/civilization/Civilization.kt

```diff
@@ -317,7 +317,7 @@ class Civilization : IsPartOfGameInfoSerialization {
 
     /** Returns only undefeated civs, aka the ones we care about */
     fun getKnownCivs() = diplomacy.values.map { it.otherCiv() }.filter { !it.isDefeated() }
-    fun knows(otherCivName: String) = diplomacy.containsKey(otherCivName)
+    fun knows(otherCivName: String) = (civName == otherCivName || diplomacy.containsKey(otherCivName))
```

Comment:
I'm not sure what the other implications of this change are, since this is a pretty central function, but the logic in this change is sound. I assume we'll discover the problems as they arise.